 mysqlc/source/mysqlc_connection.cxx |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

New commits:
commit 6b50e21473e7d2b24b5c609d8a1c31f27644d842
Author: Noel Grandin <noel@peralex.com>
Date:   Fri May 30 13:02:20 2014 +0200

    deb#749592 mysql-connector doesn't work with remote connections
    
    and also
    fdo#77584 mysql-connector doesn't work over SSH tunnel
    
    Not sure exactly how this got broken, I suspect that the UI code
    has started unconditionally passing down the properties, just filling
    them with empty spaces for the unused ones.
    
    Anyhow, this appears to fix the problem.
    
    Change-Id: I7ac2a0d6bae610f47d2a28daa9beb3ef0e2dbb52
    Reviewed-on: https://gerrit.libreoffice.org/9565
    Reviewed-by: Caolán McNamara <caolanm@redhat.com>
    Tested-by: Caolán McNamara <caolanm@redhat.com>

diff --git a/mysqlc/source/mysqlc_connection.cxx b/mysqlc/source/mysqlc_connection.cxx
index 9c0b183..f8274f4 100644
--- a/mysqlc/source/mysqlc_connection.cxx
+++ b/mysqlc/source/mysqlc_connection.cxx
@@ -158,10 +158,10 @@ void OConnection::construct(const OUString& url, const Sequence< PropertyValue >
             OSL_VERIFY( pIter->Value >>= aPass );
         } else if (pIter->Name.equalsAscii("LocalSocket")) {
             OSL_VERIFY( pIter->Value >>= sUnixSocket );
-            unixSocketPassed = true;
+            unixSocketPassed = !sUnixSocket.isEmpty();
         } else if (pIter->Name.equalsAscii("NamedPipe")) {
             OSL_VERIFY( pIter->Value >>= sNamedPipe );
-            namedPipePassed = true;
+            namedPipePassed = !sNamedPipe.isEmpty();
         } else if ( pIter->Name.equalsAscii("PublicConnectionURL")) {
             OSL_VERIFY( pIter->Value >>= m_settings.connectionURL );
         } else if ( pIter->Name.equalsAscii("NewURL")) {    // legacy name for "PublicConnectionURL"
